### DISCUSSION TITLE
Optionally Return Ranking Info of each result

### DISCUSSION THREAD

**Discussion by AymanHamdoun** [meilisearch/product#614] [Feedback & Feature Proposal]
## Feature Description

There is a very useful feature in Algolia where you can send ``` getRankingInfo = true``` in the search request which would return the ranking info (matched words, exact words, typos, etc..) of each result. 

Feel free to check their [API Reference](https://www.algolia.com/doc/api-reference/api-parameters/getRankingInfo/) 

Basically the results array would then look like this:
``` javascript
[
  {
    "id": 111,
    "title": "Some Document Title",
    "_rankingInfo": {
        "nbTypos": 0,
        "proximityDistance": 0,
        "nbExactWords": 0,
        "words": 0,
    }
  },
  ...
]
```

## How it can be helpful ? 

Assume you have several indices, each containing a certain type of entity and you want to search each of these indices and provide one list of mixed results to your users. 

Lets say you have a Movies index and a Series index. and I search for "Help". 

- My movies index returned a result titled "Help"
- My series index returned a result titled "Hell" (which is 1 typo away from the original query)

it would be very helpful to have the rankingInfo of each result, so i can simply merge the two results into one array and sort it by the ranking info attributes (matched words descending, typos descending, exact words descending, etc...) 

This is a huge part of my work actually and Algolia makes my result merging logic easy because it provides the rankingInfo for my results. If i were to compute them manually it would be such a waste of time coz the search engine already computed them at one point and i shouldn't need to compute them again. Also if i were to compute them myself, I may as well do so in a slightly different manner than the search engine which would cause some weird inconsistencies.

**Comment by gmourier** [comment]
Hey @AymanHamdoun 👋

I'm sorry for not getting back to you sooner. Thank you for all the details you share here. 

I'm locking that discussion in favor of https://github.com/meilisearch/product/discussions/594 and I will also copy/paste and link your initial message to the new place.
